core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -156,14 +165,18 @@ public void write(Iterator<Product2<K, V>> records) throws IOException {
       }
     }
 
-    File output = shuffleBlockResolver.getDataFile(shuffleId, mapId);
-    File tmp = Utils.tempFileWith(output);
-    try {
-      partitionLengths = writePartitionedFile(tmp);
-      shuffleBlockResolver.writeIndexFileAndCommit(shuffleId, mapId, partitionLengths, tmp);
-    } finally {
-      if (tmp.exists() && !tmp.delete()) {
-        logger.error("Error while deleting temp file {}", tmp.getAbsolutePath());
+    if (pluggableWriteSupport != null) {
+      partitionLengths = combineAndWritePartitionsUsingPluggableWriter();
```

Comment:
these `partitionLengths` are basically ignored, right?  Is the idea that `mapOutputWriter.commitAllPartitions();` is doing the equivalent to creating the index file in the shuffle store?

Reply:
Yeah these are ignored =/ work is needed to rip out the partition length stuff from the code paths that use the pluggable writer and delegate that stuff to, indeed, `mapOutputWriter#commitAllPartitions`.